frontendFile/src/main/java/com/virtuslab/gitmachete/frontend/file/MacheteCompletionContributor.java

```diff
@@ -0,0 +1,49 @@
+package com.virtuslab.gitmachete.frontend.file;
+
+import com.intellij.codeInsight.completion.CompletionContributor;
+import com.intellij.codeInsight.completion.CompletionParameters;
+import com.intellij.codeInsight.completion.CompletionResultSet;
+import com.intellij.codeInsight.completion.PlainPrefixMatcher;
+import com.intellij.codeInsight.completion.impl.CamelHumpMatcher;
+import com.intellij.codeInsight.lookup.LookupElementBuilder;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiFile;
+import com.intellij.ui.TextFieldWithAutoCompletionListProvider;
+import git4idea.repo.GitRepositoryManager;
+import io.vavr.collection.List;
+
+public class MacheteCompletionContributor extends CompletionContributor {
+
+  @Override
+  public void fillCompletionVariants(CompletionParameters parameters, CompletionResultSet result) {
+
+    PsiFile file = parameters.getOriginalFile();
+    Project project = file.getProject();
+
+    var gitRepository = List.ofAll(GitRepositoryManager.getInstance(project).getRepositories())
+        .find(r -> file.getVirtualFile().getPath().startsWith(r.getRoot().getPath()));
+
+    if (gitRepository.isEmpty()) {
+      return;
+    }
+
+    var lists = List.ofAll(gitRepository.get().getInfo().getLocalBranchesWithHashes().keySet())
+        .map(localBranch -> localBranch.getName());
+
+    result.stopHere();
+    int count = parameters.getInvocationCount();
+
+    String prefix = TextFieldWithAutoCompletionListProvider.getCompletionPrefix(parameters);
+    if (count == 0 && prefix.length() < 1) {
```

Comment:
`prefix.isEmpty()`

Reply:
Actually... why only return in case of `count == 0` (i.e. autopopup) but not for explicitly invoked completion? :thinking:

Reply:
...ok, I think I see the logic here, but pls drop an explicit comment anyway

Reply:
managed to get rid of and works fine